File: frontend/admin-panel/src/app/admin/types/seller.ts
// admin/types/seller.ts

export type Seller = {
  id: string;
  seller_name: string;
  seller_mobile: string;
  owner_name: string;
  email: string;
  password?: string;
  address: string;
  city: string;
  gst_number?: string;
  seller_type: string;
  plan_type?: string;
  plan_price?: number;
  payment_status: "Paid" | "Pending" | "Expired";
  renew_date: string;
  next_billing_date: string;
  trial_period_end?: string;
  last_payment_date?: string;
  subscribed_date?: string;
  max_users_allowed?: number;
  notes?: string;
  status: "Active" | "Inactive";
  created_at: string;
  updated_at: string;
};

// ✅ Ensure `password` is included in SellerCreate
export type SellerCreate = Omit<Seller, "id" | "created_at" | "updated_at"> & {
  password: string;
};

// types/seller.ts
export type SellerUpdate = {
  seller_name?: string;
  seller_mobile?: string;
  owner_name?: string;
  email?: string;
  address?: string;
  city?: string;
  gst_number?: string;
  logo_url?: string;
  seller_type?: string;
  status?: string;
  max_users_allowed?: number;
  plan_type?: string;
  plan_price?: number;
  payment_status?: string;
  subscribed_date?: string;
  renew_date?: string;
  next_billing_date?: string;
  trial_period_end?: string;
  last_payment_date?: string;
  notes?: string;
  password?: string; // Optional
};

File: backend/app/models/seller.py
from sqlalchemy import Column, String, DateTime, Numeric, JSON, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.core.database import Base
import uuid
import enum

class SellerType(str, enum.Enum):
    RETAILER = "Retailer"
    DISTRIBUTOR = "Distributor"
    MANUFACTURER = "Manufacturer"

class SellerStatus(str, enum.Enum):
    ACTIVE = "Active"
    INACTIVE = "Inactive"

class Seller(Base):
    __tablename__ = "sellers"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, index=True)
    seller_name = Column(String(255), nullable=False)
    seller_mobile = Column(String(15), nullable=False)
    owner_name = Column(String(255), nullable=False)
    email = Column(String(100), unique=True, nullable=False)  
    address = Column(String, nullable=False)
    city = Column(String(100), nullable=False)
    gst_number = Column(String(20), nullable=True)
    logo_url = Column(String(255), nullable=True)
    seller_type = Column(String, nullable=False)
    plan_type = Column(String(50), nullable=True)
    plan_price = Column(Numeric(10, 2), nullable=True)
    payment_status = Column(String, nullable=False, default="Pending")
    subscribed_date = Column(DateTime, server_default=func.now())
    renew_date = Column(DateTime, nullable=False)
    next_billing_date = Column(DateTime, nullable=False)
    trial_period_end = Column(DateTime, nullable=True)
    last_payment_date = Column(DateTime, nullable=True)
    max_users_allowed = Column(Numeric(10, 2), nullable=True)
    documents = Column(JSON, nullable=True)
    notes = Column(String, nullable=True)
    status = Column(String, nullable=False, default="Active")

    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    # ✅ Add relationship to collections
    collections = relationship("TileCollection", back_populates="seller", cascade="all, delete")
    users = relationship("User", back_populates="seller")  # Links seller to users
